src/main/java/it/pagopa/pn/delivery/middleware/notificationdao/PaNotificationLimitDaoDynamo.java

```diff
@@ -0,0 +1,113 @@
+            log.error("Unable to update item in DynamoDB: {}", e.getMessage(), e);
+            throw new PnInternalException("Update failed: " + e.getMessage(), ERROR_CODE_DELIVERY_DYNAMO_EXCEPTION);
+        }
+    }
+
+    @Override
+    public void incrementLimitDecrementDailyCounter(String paId, OffsetDateTime sentAt) {
+        log.info("Incrementing limit and decrementing dailyCounter for paId: {} sentAt: {}", paId, sentAt);
+        String dailyCounter = createDailyCounter(sentAt);
+        try {
+            UpdateItemRequest updateRequest = UpdateItemRequest.builder()
+                    .tableName(tableName)
+                    .key(createPrimaryKey(paId, sentAt))
+                    .updateExpression("ADD " + PaNotificationLimitEntity.FIELD_RESIDUAL_LIMIT + " :increment, " + dailyCounter + " :decrement")
+                    .conditionExpression("attribute_exists(" + PaNotificationLimitEntity.FIELD_PK + ")")
+                    .expressionAttributeValues(Map.of(
+                            ":increment", AttributeValue.builder().n("1").build(),
+                            ":decrement", AttributeValue.builder().n("-1").build()
+                    ))
+                    .build();
+
+            dynamoDbClient.updateItem(updateRequest);
+
+        } catch (ConditionalCheckFailedException ex) {
+            log.error("Conditional check failed: {}", ex.getMessage(), ex);
+        } catch (DynamoDbException e) {
+            log.error("Unable to update item in DynamoDB: {}", e.getMessage(), e);
+            throw new PnInternalException("Update failed: " + e.getMessage(), ERROR_CODE_DELIVERY_DYNAMO_EXCEPTION);
+        }
+    }
+
+    @Override
+    public boolean checkIfPaNotificationLimitExists(String paId, OffsetDateTime sentAt) {
+        GetItemRequest getItemRequest = GetItemRequest.builder()
+                .tableName(tableName)
+                .key(createPrimaryKey(paId, sentAt))
+                .build();
+
+        try {
+            GetItemResponse getItemResponse = dynamoDbClient.getItem(getItemRequest);
+            return getItemResponse.hasItem();
+        } catch (DynamoDbException e) {
+            log.error("Unable to get item from DynamoDB: {}", e.getMessage(), e);
+            throw new PnInternalException("GetItem failed: " + e.getMessage(), ERROR_CODE_DELIVERY_DYNAMO_EXCEPTION);
+        }
+    }
+
+    private Map<String, AttributeValue> createPrimaryKey(String paId, OffsetDateTime sentAt) {
+        String pk = paId + "##" + sentAt.getYear() + "##" + String.format("%02d", sentAt.getMonthValue());
+        log.info("Creating primary key: {}", pk);
+        return Map.of(PaNotificationLimitEntity.FIELD_PK, AttributeValue.builder().s(pk).build());
+    }
+
+    private String createDailyCounter(OffsetDateTime sentAt) {
+        String dailyCounter = DAILY_COUNTER + String.format("%02d", sentAt.getDayOfMonth());
+        log.info("Creating daily counter: {}", dailyCounter);
+        return dailyCounter;
+    }
```

Comment:
Porterei questi 2 in una classe di utils dedicata, anche allo stesso livello del Dao. Questo ci permette di testarli con Junit. In questo modo possiamo verificare i diversi casi, dunque giorno 01 giorno 10 20 31 ecc.
Allo stesso modo per il mese sopra